streams/src/main/java/org/apache/kafka/streams/state/internals/buffer/InMemoryTimeOrderedKeyValueBufferStore.java

```diff
@@ -0,0 +1,219 @@
+    private static final String BUFFER_TIME_HEADER = "_bufferTime";
+    private static final BytesSerializer KEY_SERIALIZER = new BytesSerializer();
+    private static final ByteArraySerializer VALUE_SERIALIZER = new ByteArraySerializer();
+
+    private final InMemoryTimeOrderedKeyValueBuffer buffer;
+    private final String storeName;
+
+    private RecordCollector collector;
+    private String changelogTopic;
+    private boolean open;
+
+    public static class Builder implements StoreBuilder<StateStore> {
+        private final String storeName;
+
+        public Builder(final String storeName) {
+            this.storeName = storeName;
+        }
+
+        @Override
+        public StoreBuilder<StateStore> withCachingEnabled() {
+            return null;
+        }
+
+        @Override
+        public StoreBuilder<StateStore> withCachingDisabled() {
+            return null;
+        }
+
+        @Override
+        public StoreBuilder<StateStore> withLoggingEnabled(final Map<String, String> config) {
+            return null;
+        }
+
+        @Override
+        public StoreBuilder<StateStore> withLoggingDisabled() {
+            return null;
+        }
+
+        @Override
+        public StateStore build() {
+            return new InMemoryTimeOrderedKeyValueBufferStore(
+                new InMemoryTimeOrderedKeyValueBuffer(),
+                storeName
+            );
+        }
+
+        @Override
+        public Map<String, String> logConfig() {
+            return Collections.emptyMap();
+        }
+
+        @Override
+        public boolean loggingEnabled() {
+            return true;
+        }
+
+        @Override
+        public String name() {
+            return storeName;
+        }
+    }
+
+    private InMemoryTimeOrderedKeyValueBufferStore(final InMemoryTimeOrderedKeyValueBuffer buffer,
+                                                   final String storeName) {
+        this.buffer = buffer;
+        this.storeName = storeName;
+    }
+
+    public InMemoryTimeOrderedKeyValueBuffer getBuffer() {
+        return buffer;
+    }
+
+    @Override
+    public String name() {
+        return storeName;
+    }
+
+    @Override
+    public void close() {
+        buffer.clear();
+        open = false;
+    }
+
+    @Override
+    public boolean persistent() {
+        return false;
+    }
+
+    @Override
+    public boolean isOpen() {
+        return open;
+    }
+
+    @Override
+    public void init(final ProcessorContext context, final StateStore root) {
+        context.register(root, (RecordBatchingStateRestoreCallback) this::restoreBatch);
+        try {
+            collector = ((RecordCollector.Supplier) context).recordCollector();
+            changelogTopic = ProcessorStateManager.storeChangelogTopic(context.applicationId(), storeName);
+        } catch (final UnsupportedOperationException e) {
+            LOG.warn("Disabling durability, since the context doesn't support it.", e);
```

Comment:
This happens in the unit tests.

Reply:
I am wondering if we should/can fix the test instead? Seems to be a hack to catch this.

Reply:
It is a bit hacky. How about I actually do go ahead an implement configuring logging, then the tests can disable it?

Reply:
Can you elaborate why a test needs to disable logging? Which test(s) are we taking about? Context/scope is unclear to me atm.

Reply:
The unit tests in KTableSuppressProcessorTest use the MockProcessorContext, which does not have a mocked record collector, by design. We said that if you want to use it, you have to disable logging in your stores under test.

We can change the MockProcessorContext, but it requires a KIP.

For now, that test directly instantiates the KTableSuppressProcessor and also the Buffer (using the store builder), so it's very straightforward for it to call `withLoggingDisabled` during that construction.

For the scenario test, which uses the TopologyTestDriver, and obviously for the integration tests, their contexts do support record collection, and therefore everything works fine with the default of changelogging enabled.

Reply:
SGTM. Let's add the `withLoggingDisabled` thing.